src/classes/STG_PanelDataImportAdvancedMapping_CTRL.cls

```diff
@@ -0,0 +1,277 @@
+        dataImportSettings.CMT_Deployment_ID__c = deploymentId;
+
+        try {
+            upsert dataImportSettings;
+        } catch (Exception e) {
+            ERR_Handler.processError(e, ERR_Handler_API.Context.STTG);
+            ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR, e.getMessage()));
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Action Method used by Poller to check status of CMDT deployment
+    * @return void
+    */
+    public PageReference startPolling() {
+        timeoutCount++;
+
+        deploymentResult = checkDeploymentStatus();
+        dataImportSettings = UTIL_CustomSettingsFacade.getDataImportSettings();
+
+        if (deploymentResult != null) {
+            // We had a migration that contained CMT records to insert
+            Boolean isSuccess = deploymentResult.status == Metadata.DeployStatus.Succeeded
+                && dataImportSettings.Field_Mapping_Method__c == BDI_MigrationMappingUtility.DATA_IMPORT_FIELD_MAPPING;
+
+            if (isSuccess) {
+                isPolling = false;
+                throwMessage(ApexPages.Severity.CONFIRM, Label.stgDeploymentCustomMetadataMappingEnabled);
+            } else if (deploymentResult.status == Metadata.DeployStatus.FAILED) {
+                isPolling = false;
+                throwMessage(ApexPages.Severity.ERROR, Label.stgDeploymentTryAgain);
+            }
+
+        } else if (deploymentResult == null
+                   && dataImportSettings.Field_Mapping_Method__c == BDI_MigrationMappingUtility.DATA_IMPORT_FIELD_MAPPING) {
+            // We had a migration that didn't contain CMT records to insert
+            // DeploymentContainer was empty which means we're using our standard field mapping set
+            // Org had no conflicting field mappings nor any net new field mappings in their Data Import help text
+            isPolling = false;
+            throwMessage(ApexPages.Severity.CONFIRM, Label.stgDeploymentCustomMetadataMappingEnabled);
+        }
+
+        if (timeoutCount > MAX_TIME_OUT) {
+            // We timed out, reset instance of migration mapping utility
+            migrationMappingUtility = new BDI_MigrationMappingUtility(new BDI_MigrationMappingHelper());
+            isPolling = false;
+            throwMessage(ApexPages.Severity.WARNING, Label.stgDeploymentTimedOut + ' ' + Label.stgDeploymentTryAgain);
+            return null;
+        }
+
+        return null;
+    }
+
+    /*******************************************************************************************************
+    * @description Checks to see if we have a saved custom metadata deployment id,
+    * inspects the deployment status, and sets the hasDeploymentInProgress boolean
+    * @return Metadata.DeployResult
+    */
+    private Metadata.DeployResult checkDeploymentStatus() {
```

Comment:
This method might be broken up into two methods, this one could be called 'getDeployResult', since it returns a `DeployResult' and the `hasDeploymentInProgress` member variable could be changed to an attribute that inspects the `deployResult` when called.  Not necessary, but that would allow each method/attribute to do one thing only.

If you did that, then, for example at the top of the `enableDataImportFieldMapping` method, instead of 
```
// Check the deployment status and set hasDeploymentInProgress value
        deploymentResult = checkDeploymentStatus();

        if (hasDeploymentInProgress == true) {
``` 
It could be:
```
// Get the status of the latest deployment
        setDeploymentResult();

        if (hasDeploymentInProgress == true) {
```